### PR TITLE
SHOR-26: Add missing screens for Home/Dashboard and Configure Dashboard screen

### PR DESCRIPTION
**Pull request by swastikpareek**
## PR contains
* Screen for Home Dashboard Screen
* Configure dashboard Dashboard
* Added a new group `home-dashboard` to the scenarios group list.
## Steps to test 
```shell
$ gulp backstopjs:reference --group home-dashboard && gulp backstopjs:test --group home-dashboard
```

### CODE REVIEW

File: backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+module.exports = async (engine, scenario, vp) => {
+  await engine.click('#crm-dashboard-configure');
+  // wait for the ajax to complete the DOM
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js]: unnecessary comment, the code is self-explanatory

File: backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+module.exports = async (engine, scenario, vp) => {
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js]: if you don't need `scenario` and `vp`, then don't list them

File: backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+'use strict';
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js]: just call the folder `dashboard`, `home-dashboard` is overkill

File: backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+module.exports = async (engine, scenario, vp) => {
+  await engine.click('#crm-dashboard-configure');
+  // wait for the ajax to complete the DOM 
+  await engine.waitFor('#configure-dashlet', { visible: true });
+  //wait for 500ms till the animation completes
+  await engine.waitFor(500);
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js]: what animation? is there anything we can do to avoid this arbitrary number?

File: backstop_data/engine_scripts/puppet/mailings/ab-test-manage.js [hunk 5]
```diff
@@ -3,8 +3,8 @@
 module.exports = async (engine, scenario, vp) => {
   await engine.waitFor('table.display', { visible: true });
 
-  // The list grows on each passing run, so we display only 10 rows in the test
+  // The list grows on each passing run, so we display only 9 rows in the test (First being the main row);
   await engine.evaluate(() => {
-    document.querySelectorAll('table.display tr:nth-child(n+10)').forEach(row => row.remove());
+    document.querySelectorAll('table.display tr:nth-child(n+4)').forEach(row => row.remove());
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/mailings/ab-test-manage.js]: why was this change necessary at all?

File: scenarios/home-dashboard.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (AkA84) [inline scenarios/home-dashboard.json]: call this file `dashboard.json`

File: scenarios/home-dashboard.json [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  "scenarios": [{
+    "label": "Dashboard",
+    "url": "{url}/civicrm/dashboard"
+  },
+  {
+    "label": "Dashboard - Configure",
+    "url": "{url}/civicrm/dashboard",
+    "onReadyScript": "home-dashboard/configure-dashboard.js"
+  }]
+}
```
Comment (AkA84) [inline scenarios/home-dashboard.json]: make sure to configure your IDE to leave an empty line on save

File: backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js (hunk 2 above)
Comment (swastikpareek) [reply backstop_data/engine_scripts/puppet/home-dashboard/configure-dashboard.js]: I checked at other places to fix this but these 3 params are passed in each function in every file. I guess to maintain a standard we can have these 3 params in the function.

File: backstop_data/engine_scripts/puppet/mailings/ab-test-manage.js (hunk 5 above)
Comment (swastikpareek) [reply backstop_data/engine_scripts/puppet/mailings/ab-test-manage.js]: Since in the Vanilla CiviCRM there are too few fields and was creating a problem while testing the scenario. 
What is your opinion?

File: backstop_data/engine_scripts/puppet/dashboard/configure-dashboard.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+module.exports = async (engine) => {
+  await engine.click('#crm-dashboard-configure');
+  await engine.waitFor('#configure-dashlet', { visible: true });
+  await engine._client.send('Animation.setPlaybackRate', { playbackRate: 2 })
```
Comment (AkA84) [inline backstop_data/engine_scripts/puppet/dashboard/configure-dashboard.js]: what's this? what does it do?

File: backstop_data/engine_scripts/puppet/mailings/ab-test-manage.js (hunk 5 above)
Comment (AkA84) [reply backstop_data/engine_scripts/puppet/mailings/ab-test-manage.js]: I'm not sure i understand.

You changed the comment from "only 10 rows" to "only 9 rows", but then you changed the selector to remove any row past the 4th one.

And I don't understand what you mean by "too few fields". What fields? What sort of problems were you encountering?

File: backstop_data/engine_scripts/puppet/dashboard/configure-dashboard.js (hunk 8 above)
Comment (swastikpareek) [reply backstop_data/engine_scripts/puppet/dashboard/configure-dashboard.js]: So, basically when we click on *Configure dashboard* the edit dashboard screen opens with a fade in animation. So the screenshot captured some times doesn't match as they are translucent (fade in effect not completed at time of screenshot).
So this code virtually fast forward the screenshot process (by increasing the playback Rate) after the animation.
Link to reference - https://github.com/GoogleChrome/puppeteer/issues/453
